### PR TITLE
helix: add module

### PR DESCRIPTION
**Pull request by Philipp-M**
### Description

Adds helix-editor as module supporting the options settings, languages and themes and package.

This currently requires (and is tested with) helix-editor from master (using the current https://docs.helix-editor.com/ as reference), which is why I marked this as a draft (I'm not sure what helix-editor 0.5 supports, and it's not officially documented AFAIK)

Apart from this version issue, this PR is ready to merge.
I'm not sure if it makes sense to merge this as is, or to wait for a new release of helix.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by Philipp-M** [comment]
Just now a new release occured :)

So I guess this is ready for review/merge

File: modules/programs/helix.nix [hunk 1]
```diff
@@ -0,0 +1,150 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.helix;
+  tomlFormat = pkgs.formats.toml { };
+in {
+  meta.maintainers = [ maintainers.Philipp-M ];
```
Comment (berbiche) [inline modules/programs/helix.nix:9]:
```suggestion
  meta.maintainers = [ hm.maintainers.Philipp-M ];
```

File: modules/programs/helix.nix [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+  options.programs.helix = {
+    enable = mkEnableOption "helix";
```
Comment (berbiche) [inline modules/programs/helix.nix:12]:
```suggestion
    enable = mkEnableOption "helix text editor";
```

File: modules/programs/helix.nix [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            "debug" = dark-gray;
+            "warning" = yellow;
+            "error" = red;
+          };
+        }
+      '';
+      description = ''
+        Themes written to
+        <filename>$XDG_CONFIG_HOME/helix/themes/*.toml</filename>.
+        </para><para>
```
Comment (berbiche) [inline modules/programs/helix.nix:124]: Maybe it should specify the name of the attribute is the name of the theme/file.

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: modules/programs/helix.nix (hunk 3 above)
Comment (Philipp-M) [reply modules/programs/helix.nix]: Yeah I wasn't too happy with the description as well, is the current version better?

**Comment by colemickens** [comment]
This works for me, thanks, +1 for merging after the tests are fixed.

**Comment by Philipp-M** [comment]
Thanks, AFAIK the failing tests don't have anything to do with my PR, they existed on master at that time, I quickly rebase onto master, and I hope they run through then

**Review by colemickens — APPROVED** [review]

**Comment by rycee** [comment]
Thanks! Merged to master now. The `themes` example is a bit larger than I usually prefer so maybe can try abbreviating it at some point in the future.

**Comment by Philipp-M** [comment]
Thanks!
Yeah tried that first, but helix had some missing theme variables, I then took a complete theme example instead, but I will look into it, if it's possible to strip some of the theme variables

**Pull request merged**